compiler/rustc_middle/src/ty/query/profiling_support.rs

```diff
@@ -66,17 +66,25 @@ impl<'p, 'c, 'tcx> QueryKeyStringBuilder<'p, 'c, 'tcx> {
                 end_index = 3;
             }
             other => {
-                name = other.as_symbol();
-                if def_key.disambiguated_data.disambiguator == 0 {
-                    dis = "";
-                    end_index = 3;
-                } else {
-                    write!(&mut dis_buffer[..], "[{}]", def_key.disambiguated_data.disambiguator)
+                name = match other.get_name() {
+                    DefPathDataName::Named(name) => {
+                        dis = "";
+                        end_index = 3;
+                        name
+                    }
+                    DefPathDataName::Anon { namespace } => {
+                        write!(
+                            &mut dis_buffer[..],
+                            "[{}]",
+                            def_key.disambiguated_data.disambiguator
+                        )
                         .unwrap();
-                    let end_of_dis = dis_buffer.iter().position(|&c| c == b']').unwrap();
-                    dis = std::str::from_utf8(&dis_buffer[..end_of_dis + 1]).unwrap();
-                    end_index = 4;
-                }
+                        let end_of_dis = dis_buffer.iter().position(|&c| c == b']').unwrap();
+                        dis = std::str::from_utf8(&dis_buffer[..end_of_dis + 1]).unwrap();
+                        end_index = 4;
+                        namespace
```

Comment:
Hmm I think this would result in `closure[123]` instead of `{{closure}}[123]` (before) or `{closure#123}` (that we want).

cc @wesleywiser @Mark-Simulacrum Do we have no way of testing profiling with query keys? It would be nice to have a before/after in this PR. But I guess bringing in e.g. `crox` (or something more custom based on `analyzeme`) would not be cheap (to build in the first place, I mean)? (unless we can reuse all of the `rustc`/tools dependencies? since I think most are already used by something else)

Reply:
Should be fixed now